src/screens/Customers/AddCustomer.tsx

```diff
@@ -1,38 +1,116 @@
+  currentSite: SiteRecord;
 }
 
+
 function AddCustomer(props: AddCustomerProps) {
-  const { classes } = props;
+  const { classes, currentSite } = props;
+
+  const [selectedTariffPlan, setSelectedTariffPlan] = useState("");
+  const [customerName, setCustomerName] = useState("");
+  const [meter, setMeter] = useState("");
+  const [meterChecked, setMeterChecked] = useState(false);
+  const [customerInactive, setCustomerInactive] = useState(false);
+
+  const handleSelectTariffPlan = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setSelectedTariffPlan(event.target.value as string);
+  }
+
+  const handleSubmit = (event: React.MouseEvent) => {
+    // Prevent page refresh on submit
+    event.preventDefault();
+
+    const customer = EMPTY_CUSTOMER;
+    customer.name = customerName;
+    customer.meterNumber = parseInt(meter);
+    customer.isactive = !customerInactive;
+    customer.tariffPlansId = selectedTariffPlan;
+    addCustomer(customer);
+    
+    const cust = {
+      ...customer,
+      sitesId: currentSite.rid
+    }
+    
+    createCustomer(cust);
+  }
+
+  const handleNameInput = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setCustomerName(event.target.value as string);
+  }
+
+  const handleMeterInput = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setMeter(event.target.value as string);
+  }
+
+  const tariffPlans = currentSite.tariffPlans;
 
   return (
     <BaseScreen title="Add New Customer" leftIcon="backNav">
-      <form noValidate className={classes.content}>
-        <TextField label={'Name'} />
-        <Checkbox label={'Select if customer is inactive'} />
-        <Checkbox label={'Meter:'} textField={'meter'} />
-        <TextField label={'Tariff Plan'} />
-        <Button label={'ADD'} />
+      <form noValidate className={classes.content} onSubmit={() => false}>
+        <TextField label={'Name'} id={'name'} primary={true} onChange={handleNameInput} />
+        <FormControl variant="outlined" className={classes.formControl}>
+          <InputLabel id="select-tariff-plan-label">Select Tariff Plan</InputLabel>
+          <Select label={"Select Tariff Plan"} id={'select-tariff-plan'} labelId = "select-tariff-plan-label" onChange={handleSelectTariffPlan}>
+            {tariffPlans.map((plan, index) =>
+                <MenuItem key={index} value={plan.id}>{plan.name}</MenuItem>
+            )}
+          </Select>
+        </FormControl>
+        <Checkbox label={'Meter:'} textField={meterChecked ? 'meter': null} checkboxOnChange={() => setMeterChecked(!meterChecked)}textFieldOnChange={handleMeterInput}/>
+        <Checkbox label={'Customer is inactive'} checkboxOnChange = {() => setCustomerInactive(!customerInactive)}/>
+        <Button label={'Add'} onClick={(event) => handleSubmit(event)} />
       </form>
     </BaseScreen>
   );
 }
 
-export default withStyles(styles)(AddCustomer);
+const mapStateToProps = (state: RootState) => ({
+  currentSite: state.siteData.currentSite
```

Comment:
Right now, `currentSite` is defaulted to none (I'm going to change this in a PR to just default to EMPTY_SITE). There shouldn't be a case where it ends up null here, but better to write defensive code for now in case that changes. Can you do `... || EMPTY_SITE`?